pretyping/typeclasses.ml

```diff
@@ -31,6 +31,7 @@ type 'a hint_info_gen =
       hint_pattern : 'a option }
 
 type hint_info_expr = Constrexpr.constr_pattern_expr hint_info_gen
```

Comment:
I think `hint_info_expr` should be moved to `tactics/hints` then.

Reply:
>I think hint_info_expr should be moved to tactics/hints then.

You did that in some commit in #7570 but it seems like it depends on some previous stuff, what do?

Reply:
I think I had checked that it could be done in the context of this PR, but I was likely wrong.

If my request is not possible please dismiss my review and merge.